tt_metal/hw/inc/risc_attribs.h

```diff
@@ -15,8 +15,13 @@ union tt_uint64_t {
     };
 };
 
-#define tt_l1_ptr __attribute__((rvtt_l1_ptr))
-#define tt_reg_ptr __attribute__((rvtt_reg_ptr))
+#if defined(KERNEL_BUILD) || defined(FW_BUILD)
```

Comment:
this file shouldn't be included on host, I prefer a change higher up that precludes that.
and yes, much of this needs a rethink...